### PR TITLE
[SYCL] Do not set CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON for plugins

### PR DESCRIPTION
**Pull request by smaslov-intel**
We just don't need it:
1) PI interfaces are specifically marked with __SYCL_EXPORT, e.g. https://github.com/intel/llvm/blob/8e0cc4b7a845df9389a1313a3e680babc4d87782/sycl/include/sycl/detail/pi.h#L1148
2) We call to PI interfaces (except for `piPluginInit`) via a pointer from a table set up during `piPluginInit`:https://github.com/intel/llvm/blob/8e0cc4b7a845df9389a1313a3e680babc4d87782/sycl/include/sycl/detail/pi.hpp#L220

### CODE REVIEW

**Comment by smaslov-intel** [comment]
Waiting on code owner review from @intel/dpcpp-l0-pi-reviewers and/or @intel/llvm-reviewers-runtime.

**Review by aelovikov-intel — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Pull request merged**
